doc/userManual.md

```diff
@@ -0,0 +1,183 @@
+
+The following is a brief description of the main VIs you have to use to work with **commands** in the TCP server, and an example how to implement the server.
+
+## getInstanceCmd&#46;vi
+
+The getInstanceCmd VI is used to get the `CmdRead` user event instance, to read commands from TCP client.
+
+![getInstanceCmd](./images/getInstanceCmd.png)
+
+## regCmd&#46;vi
+
+The regCmd VI is used to inform the TCP server which commands are registered and allowed to be used by the LabVIEW component.
+Please define all of the registered commands in the `cmdReg` 1-D array of strings.
+
+![regCmd](./images/regCmd.png)
+
+## sendCmdStatus&#46;vi
+
+The sendCmdStatus VI is used to send a `cmdStatus` message to the TCP server to inform the TCP client about the command result.
+You need to add the `sequence_id` of the command received as an input of this function.
+
+![sendCmdStatus](./images/sendCmdStatus.png)
+
+The following image shows an example of a TCP server using the library to receive commands from a TCP client.
+
+Notice that this program configures the TCP server, registers the allowed commands (`move` and `start` in this example), gets the user event reference to receive the commands, runs, and stops the TCP server.
+
+This example registers the `CmdRead` user event and waits for it.
+When the command is received it is showed in the `Command Received` indicator.
+
+![exampleCommand](./images/exampleCommand.png)
+
+After receiving the commands you can send the result of the execution, using the **sendCmdStatus.vi** with the `sequence_id` of the received command.
+
+## Using the TcpServer to receive and send events
+
+The following is a brief description of the main VIs you have to use to work with **events** in the TCP server, and an example how to implement the server.
+
+## regEvt&#46;vi
+
+The regEvt VI is used to inform the TCP server which events are registered and allowed to be processed.
+Please define all of the registered events in the `evtReg` 2-D array of strings, using the following structure:
+
+| Component Name | Event Name  |
+|:--------------:|:-----------:|
+|  component 1   |   event 1   |
+|  component 2   |   event 2   |
+
+![regEvt](./images/regEvt.png)
+
+## getInstanceEvt&#46;vi
+
+The getInstanceEvt VI is used to get the `EvtRead` user event instance, to receive an event from the TCP client.
+
+![getInstanceEvt](./images/getInstanceEvt.png)
+
+The following image shows an example of a TCP server using the library to receive events.
+
+Notice that this program configures the TCP server, registers the allowed events, gets the user event reference to receive the events, and runs the TCP server.
+
+This example registers the `EvtRead` user event and waits for it.
+When the event is received it is showed in the `Event Received` indicator.
+
+![exampleEvent](./images/exampleEvent.png)
+
+If you need to send an event to the TCP client use the **sendEvt.vi** function as the following images shows.
```

Comment:
Please mention we have a vi (**convertToKeyValuePair.vi**) to help the user write the JSON string easily. Add a hyperlink to the use of this polymorphic vi.

Reply:
Done.